service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -390,4 +392,30 @@ public String enablePetServiceAccountImpersonation(
       throw new RuntimeException("Error enabling user's pet SA", e);
     }
   }
+
+  public void removeRole(
+      UUID workspaceId, WsmIamRole role, String userEmail, AuthenticatedUserRequest userRequest) {
+    Workspace workspace =
+        validateWorkspaceAndAction(userRequest, workspaceId, SamConstants.SAM_WORKSPACE_OWN_ACTION);
+    stageService.assertMcWorkspace(workspace, "deleteGcpCloudContext");
+    // Before launching the flight, validate that the user being removed is a direct member of the
+    // specified role. Users may also be added to a workspace via managed groups, but WSM does not
+    // control membership of those groups, and so cannot remove them here.
+    List<String> roleMembers =
+        samService.listWorkspacePolicyMembers(workspaceId, role, userRequest);
+    if (!roleMembers.contains(userEmail)) {
+      return;
+    }
+    jobService
+        .newJob(
+            "Remove user from workspace " + workspaceId,
```

Comment:
aren't we only removing the user from the workspace if this is their last role?

Reply:
good point, updated description.